woocommerce/woocommerce-paypal-payments
# Functionality to choose subscription failure behavior

**Issue**: #728

---

### Description

We needed a functionality to modify the behavior for failed subscription payments.
Currently, these authorizations would be voided and the orders would fail after three minutes.
Some merchants may prefer orders to instead be captured and then the subscription set to “Manual renewal”.
For example, when the buyer has multiple products in the cart, they may also not care too much about the renewal compared to the initial payment.

The PR will add the following functionalities:

- In the PayPal Checkout Settings section and in the PayPal Card Processing settings, display a new setting called _Subscription capture behavior if Vault fails_
- It has a dropdown menu, containing these two settings (only one can be selected at a time):

Case | Setting name | Order note | Customer email text
-- | -- | -- | --
if the payment method can’t be saved, fail the order and void the auth (current behavior) | void authorization & fail the order/subscription | Subscription payment failed because the payment method could not be saved at PayPal. Contact PayPal MTS for guidance. | The subscription payment failed because the payment method could not be saved. Please try again with a different payment method.
if the payment method can’t be saved, capture the payment anyway, and set the subscription to Manual Renewal | capture authorized payment & set subscription to Manual Renewal | Subscription set to Manual Renewal because the payment method could not be saved at PayPal | Subscription set to Manual Renewal because the payment method could not be saved. To enable automatic renewal payments, please try a different payment method

![6235ae60-f5c8-4e0e-af4b-25ee077da8cd](https://user-images.githubusercontent.com/11319597/178263611-0a60209c-8c9e-4937-94a2-b6a467341d75.png)

![00edaf63-10c9-4728-8b67-18c454f83526](https://user-images.githubusercontent.com/11319597/178263635-064b30f6-b6f3-4f4f-a7eb-b3d9ff2bd54f.png)

### Steps to Test




1. Enable vaulting.
2. Checkout for subscription product.
3. Make the vaulting to fail
4. See the order behavior.

---

Closes #728 .
